expr/func_sortby_test.go

```diff
@@ -301,29 +132,32 @@ func TestSortByMaxReverseLong(t *testing.T) {
 func testSortBy(name string, fn string, reverse bool, in []models.Series, out []models.Series, t *testing.T) {
 	f := NewSortByConstructor(fn, reverse)()
 	f.(*FuncSortBy).in = NewMock(in)
-	gots, err := f.Exec(make(map[Req][]models.Series))
-	if err != nil {
-		t.Fatalf("case %q: err should be nil. got %q", name, err)
-	}
-	if len(gots) != len(out) {
-		t.Fatalf("case %q: len output expected %d, got %d", name, len(out), len(gots))
+
+	// Copy input to check that it is unchanged later
+	inputCopy := make([]models.Series, len(in))
+	copy(inputCopy, in)
+
+	dataMap := DataMap(make(map[Req][]models.Series))
+
+	got, err := f.Exec(dataMap)
+	if err := equalOutput(out, got, nil, err); err != nil {
+		t.Fatalf("Case %s: %s", name, err)
 	}
-	for i, g := range gots {
-		exp := out[i]
-		if g.QueryPatt != exp.QueryPatt {
-			t.Fatalf("case %q: expected target %q, got %q", name, exp.QueryPatt, g.QueryPatt)
-		}
-		if len(g.Datapoints) != len(exp.Datapoints) {
-			t.Fatalf("case %q: len output expected %d, got %d", name, len(exp.Datapoints), len(g.Datapoints))
-		}
-		for j, p := range g.Datapoints {
-			bothNaN := math.IsNaN(p.Val) && math.IsNaN(exp.Datapoints[j].Val)
-			if (bothNaN || p.Val == exp.Datapoints[j].Val) && p.Ts == exp.Datapoints[j].Ts {
-				continue
+
+	/*
+		TODO - Is sorting modification?
```

Comment:
What is your opinion on this? I think it has to be, since 2 different branches could sort in 2 different ways and we wouldn't want them to clobber each other.

Reply:
yeah. ordering may be implied, but relied upon in unpredictable ways.
let's sort into a new output slice, then we can enable this test.